windows.go

```diff
@@ -360,8 +373,11 @@ func (w *Watcher) startRead(watch *watch) error {
 		w.mu.Unlock()
 		return nil
 	}
-	e := syscall.ReadDirectoryChanges(watch.ino.handle, &watch.buf[0],
-		uint32(unsafe.Sizeof(watch.buf)), w.recursive, mask, nil, &watch.ov, 0)
+
+	hdr := (*reflect.SliceHeader)(unsafe.Pointer(&watch.buf))
```

Comment:
Not sure what's going on here 😅

Reply:
In Go, a slice is implemented as
- a pointer to some memory
- a length field
- a capacity field

https://golang.org/pkg/reflect/#SliceHeader

The windows syscall is all C. It needs a pointer to memory. So we unpack the slice, and pull the memory pointer out.

it's conceptually not that hard, but Go REALLY doesn't want you to do this (because it starts to mess with the Go garbage collector), so the syntax is pretty awkward

Reply:
ahh got it, that makes sense!